### PR TITLE
Fix umbrella actions to dynamically support single account 

### PR DESCRIPTION
**Pull request by ravikiranvm**
Fixes OPS-2384.

## Testing Checklist

Check all that apply:

- [ ] I tested the feature thoroughly, including edge cases

- [ ] I verified all affected areas still work as expected

- [ ] Automated tests were added/updated if necessary

- [ ] Changes are backwards compatible with any existing data, otherwise a migration script is provided

## Visual Changes (if applicable)

If there are UI/UX changes, include at least one of the following:

- Screenshots
- Loom video
- Preview deployment link

### CODE REVIEW

File: packages/blocks/anodot/src/lib/get-recommendations-action-custom.ts [hunk 1]
```diff
@@ -120,8 +129,8 @@ export const getRecommendationsCustomAction = createAction({
       for (const selectedAccount of accounts) {
         const accountApiKey = buildUserAccountApiKey(
           anodotTokens.apikey,
-          selectedAccount.accountKey,
-          selectedAccount.divisionId,
+          String(selectedAccount.accountKey),
```
Comment (bigfluffycookie) [inline packages/blocks/anodot/src/lib/get-recommendations-action-custom.ts:132]: hmm why is this needed?

File: packages/blocks/anodot/test/get-recommendations-action-custom.test.ts [hunk 2]
```diff
@@ -302,6 +302,87 @@ describe('getRecommendationsCustomAction', () => {
     );
   });
 
+  test.each([
+    // Array of accounts
```
Comment (bigfluffycookie) [inline packages/blocks/anodot/test/get-recommendations-action-custom.test.ts]: please remove comments.

Comment (ravikiranvm) [reply packages/blocks/anodot/test/get-recommendations-action-custom.test.ts]: Removed

File: packages/blocks/anodot/src/lib/get-recommendations-action-custom.ts (hunk 1 above)
Comment (ravikiranvm) [reply packages/blocks/anodot/src/lib/get-recommendations-action-custom.ts:132]: buildUserAccountApiKey expects accountKey and divisionId to be string. 
```
export function buildUserAccountApiKey(
  apikey: string,
  accountKey: string,
  divisionId: string,
): string {
  const sanitizedApiKey = apikey.replace(/:-.*$/, '');

  return `${sanitizedApiKey}:${accountKey}:${divisionId}`;
}
```
but I declared them as number in 
```
export type SelectedAccount = {
  accountKey: number;
  divisionId: number;
  accountName: string;
};
```
So I converted them to String
